From 6b3201b743858a97c60bcf50baf25dc19131e248 Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Fri, 5 Sep 2014 16:36:29 -0400
Subject: [target] iscsi: Explicily clear login response PDU in exception path

Message-id: <1409934989-29941-32-git-send-email-agrover@redhat.com>
Patchwork-id: 91476
O-Subject: [RHEL7.1 PATCH e-stor 31/31] iscsi-target: Explicily clear login response PDU in exception path
Bugzilla: 1129387
RH-Acked-by: Chris Leech <cleech@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Nicholas Bellinger <nab@linux-iscsi.org>

commit 683497566d48f86e04d026de1ee658dd74fc1077 upstream.

This patch adds a explicit memset to the login response PDU
exception path in iscsit_tx_login_rsp().

This addresses a regression bug introduced in commit baa4d64b
where the initiator would end up not receiving the login
response and associated status class + detail, before closing
the login connection.

Reported-by: Christophe Vu-Brugier <cvubrugier@yahoo.fr>
Tested-by: Christophe Vu-Brugier <cvubrugier@yahoo.fr>
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/target/iscsi/iscsi_target_util.c b/drivers/target/iscsi/iscsi_target_util.c
index 77dad24..c9790f6 100644
--- a/drivers/target/iscsi/iscsi_target_util.c
+++ b/drivers/target/iscsi/iscsi_target_util.c
@@ -1288,6 +1288,8 @@ int iscsit_tx_login_rsp(struct iscsi_conn *conn, u8 status_class, u8 status_deta
  login->login_failed = 1;
  iscsit_collect_login_stats(conn, status_class, status_detail);
 
+ memset(&login->rsp[0], 0, ISCSI_HDR_LEN);
+
  hdr = (struct iscsi_login_rsp *)&login->rsp[0];
  hdr->opcode  = ISCSI_OP_LOGIN_RSP;
  hdr->status_class = status_class;
-- 
1.7.1